### DISCUSSION TITLE
Prevent removal of cusotm values in the .camel.yaml file.

### DISCUSSION THREAD

**Discussion by ImanEmadi** [apache/camel-karavan#1238] [Q&A] [answered]
### Ask a question

Hi,
i have added a new tab to the extension for myself, where it adds new values to the .camel.yaml file.
the problem is that, every time i reopen the file, or switch from a tab to the web view, the extension is apparently validating the yaml file and removes any property i have added to it.

I could not find where the logic for this functionality is.
how can i either define keywords that should not be removed from the file, or where is the logic that validates and removes unknown properties ?

### Variant

Vs Code Extension

### Container Management (if applicable)

None

### Operating System (if applicable)

Linux

### Version

4.5.0

**Comment by mgubaidullin** [comment]
What do you mean unknown properties?
Karavan supports Camel Yaml DSL. You can not add you own scuff in xxx.camel.yaml

**Comment by ImanEmadi** [comment]
I created a new tab, and there i am adding references to the .camel.yaml like this :
```yaml
- reference:
   id: 'test'
   value: 'test value'
```

the only problem is that, the extension removes it from the file whenever i open the file.
where in the project can i add an exception to it, so it keeps this part of the file ?

**Reply by mgubaidullin** [reply, marked as answer]
There is no `reference` in Camel Yaml dsl

**Reply by ImanEmadi** [reply]
I know, how can i define it by myself ? where in the project is it reading the file and removing the properties that are not part of the camel yaml dsl ?

**Reply by davsclaus** [reply]
You can have your own yaml files that are not camel, eg foo.yaml, and have whatever stuff you want there. 
karavan edits xxx.camel.yaml files

**Reply by ImanEmadi** [reply]
I know, but i want to add a feature to karavan, for personal use, i need to override the behavior so i can add new elements to the .camel.yaml which are not yaml dsl. **i am simply asking for the file that is responsible for validating and modifying the file**, so i can override the behavior. i am trying not to contribute to the karavan repo, this is something for our own needs.

**Reply by mgubaidullin** [reply]
https://github.com/apache/camel-karavan/blob/78f2353e541d425df7444a3001e8eecd85eb5a57/karavan-core/src/core/api/CamelDefinitionYaml.ts

**Reply by ImanEmadi** [reply]
Thank you!
